benchmarks/uhs_leveldb.cpp

```diff
@@ -0,0 +1,203 @@
+        opt.create_if_missing = true;
+
+        auto mint_tx1 = wallet1.mint_new_coins(2, 100);
+        auto mint_tx2 = wallet2.mint_new_coins(1, 100);
+        wallet1.confirm_transaction(mint_tx1);
+        wallet2.confirm_transaction(mint_tx2);
+
+        res = leveldb::DB::Open(opt, g_shard_test_dir, &db_ptr);
+        m_db.reset(db_ptr);
+
+        block.push_back(cbdc::transaction::compact_tx(mint_tx1));
+        block.push_back(cbdc::transaction::compact_tx(mint_tx2));
+
+        m_valid_tx
+            = wallet1.send_to(100, wallet2.generate_key(), true).value();
+        block.push_back(cbdc::transaction::compact_tx(m_valid_tx));
+        block_abridged.push_back(cbdc::transaction::compact_tx(m_valid_tx));
+
+        for(int i = 0; i < 10; i++) {
+            m_valid_tx
+                = wallet1.send_to(100, wallet2.generate_key(), true).value();
+            wallet1.confirm_transaction(m_valid_tx);
+            wallet2.confirm_transaction(m_valid_tx);
+            block.push_back(cbdc::transaction::compact_tx(m_valid_tx));
+
+            m_valid_tx
+                = wallet2.send_to(50, wallet1.generate_key(), true).value();
+            wallet1.confirm_transaction(m_valid_tx);
+            wallet2.confirm_transaction(m_valid_tx);
+            block.push_back(cbdc::transaction::compact_tx(m_valid_tx));
+
+            m_valid_tx
+                = wallet2.send_to(50, wallet1.generate_key(), true).value();
+            wallet1.confirm_transaction(m_valid_tx);
+            wallet2.confirm_transaction(m_valid_tx);
+            block.push_back(cbdc::transaction::compact_tx(m_valid_tx));
+        }
+    }
+
+    void tear_down() {
+        std::filesystem::remove_all(g_shard_test_dir);
+    }
+};
+
+// test placing a new element in DB
+static void uhs_leveldb_put_new(benchmark::State& state) {
+    auto db = db_container();
+
+    // simulate storage of a new set of transactions
+    for(auto _ : state) {
+        state.PauseTiming();
+        db.m_valid_tx
+            = db.wallet2.send_to(50, db.wallet1.generate_key(), true).value();
+        db.wallet1.confirm_transaction(db.m_valid_tx);
+        db.wallet2.confirm_transaction(db.m_valid_tx);
+        db.m_valid_tx
+            = db.wallet1.send_to(50, db.wallet2.generate_key(), true).value();
+        db.wallet1.confirm_transaction(db.m_valid_tx);
+        db.wallet2.confirm_transaction(db.m_valid_tx);
+
+        db.m_cp_tx = cbdc::transaction::compact_tx(db.m_valid_tx);
+        std::array<char, sizeof(db.m_cp_tx.m_uhs_outputs)> out_arr{};
+        std::memcpy(out_arr.data(),
+                    db.m_cp_tx.m_uhs_outputs.data(),
+                    db.m_cp_tx.m_uhs_outputs.size());
+        leveldb::Slice OutPointKey(out_arr.data(),
+                                   db.m_cp_tx.m_uhs_outputs.size());
+
+        // actual storage
+        state.ResumeTiming();
+        db.res = db.m_db->Put(db.write_opt, OutPointKey, leveldb::Slice());
+        state.PauseTiming();
```

Comment:
Here, and below: Are these end-of-loop `::PauseTiming()` calls necessary? You have one at the beginning of each loop as well. Given that the call comes off `state` which is what's being looped over (and is thus persisting across iterations), I'd imagine this (at most) might skip timing the loop iteration mechanics. If so, should we do that everywhere?

Reply:
Ah this is an oversight, I intended to remove the `::PauseTiming() `calls at the start of the loop